### PR TITLE
Support Management API and flow statistics

### PR DESCRIPTION
**Pull request by fuyufjh**
Enable manager API by specifying and IP address:

```
shadowsocks-server --manager-address 127.0.0.1:6001
```

Protocol:

https://github.com/shadowsocks/shadowsocks/wiki/Manage-Multiple-Users#protocol

### CODE REVIEW

**Comment by gyteng** [comment]
will this feature be merge to master branch?

**Comment by weaselpp** [comment]
It's very useful, could you merge to master branch?

**Comment by iineva** [comment]
It's cool!

**Comment by arthurkiller** [comment]
LGTM

thx for contribute

File: cmd/shadowsocks-server/server.go [hunk 1]
```diff
@@ -451,5 +486,113 @@ func main() {
...
+	port := parsePortNum(params.ServerPort)
+	if port == "" {
+		return []byte("err")
+	}
+	log.Printf("closing port %s\n", port)
+	passwdManager.del(port)
+	return []byte("ok")
+}
+
+func handlePing() []byte {
```
Comment (gyteng) [inline cmd/shadowsocks-server/server.go]: handlePing function must return 'pong' as the management protocol

Comment (arthurkiller) [reply cmd/shadowsocks-server/server.go]: aha, I see that

**Pull request merged**